lib.sh

```diff
@@ -440,34 +440,45 @@ fi
 # https://certbot.eff.org/docs/using.html#certbot-command-line-options
 generate_cert() {
 uir_hsts=""
+disable_000=""
 if [ -z "$SUBDOMAIN" ]
 then
     uir_hsts="--uir --hsts"
 fi
+if [ -e "$SITES_AVAILABLE"/../sites-enabled/000-default.conf ]
+then
+    a2dissite 000-default.conf
+    service apache2 reload
+    local  disable_000="yes"
+fi
 default_le="--rsa-key-size 4096 --renew-by-default --no-eff-email --agree-tos $uir_hsts --server https://acme-v02.api.letsencrypt.org/directory -d $1"
-a2dissite 000-default.conf
-service apache2 reload
 #http-01
 local  standalone="certbot certonly --standalone --pre-hook 'service apache2 stop' --post-hook 'service apache2 start' $default_le"
 #tls-alpn-01
 local  tls_alpn_01="certbot certonly --preferred-challenges tls-alpn-01 $default_le"
 #dns
 local  dns="certbot certonly --manual --manual-public-ip-logging-ok --preferred-challenges dns $default_le"
 local  methods=(standalone dns)
-for f in "${methods[@]}"
+
+for f in ${methods[*]}
 do
     print_text_in_color "${ICyan}" "Trying to generate certs and validate them with $f method."
-    eval  ff=$(echo "\$$f")
+    ff="\$$f"
```

Comment:
I don't know if this is the bug, but current status is:

```
+ methods=(standalone dns)
+ local methods
+ for f in ${methods[*]}
+ print_text_in_color '\e[0;96m' 'Trying to generate certs and validate them with standalone method.'
+ printf '%b%s%b\n' '\e[0;96m' 'Trying to generate certs and validate them with standalone method.' '\e[0m'
Trying to generate certs and validate them with standalone method.
+ ff='$standalone'
+ eval '$standalone'
++ certbot certonly --standalone --pre-hook ''\''service' apache2 'stop'\''' --post-hook ''\''service' apache2 'start'\''' --rsa-key-size 4096 --renew-by-default --no-eff-email --agree-tos --uir --hsts --server https://acme-v02.api.letsencrypt.org/directory -d cloud.ellahansson.se
usage: 
  certbot [SUBCOMMAND] [options] [-d DOMAIN] [-d DOMAIN] ...

Certbot can obtain and install HTTPS/TLS/SSL certificates.  By default,
it will attempt to use a webserver both for obtaining and installing the
certificate. 
certbot: error: unrecognized arguments: apache2 stop' apache2 start'
```

Standalone doesn't work.

Reply:
I will do a test for this.

Reply:
@enoch85 
 
I have done a test, it works well with dns and standalone.  Maybe Your code is not up to date.

Reply:
@enoch85 

The reason is that the variable does not expand correctly。I have fixed it.